e2e/nomostest/testpredicates/predicates.go

```diff
@@ -1155,6 +1156,35 @@ func validateRootSyncCondition(actual *v1beta1.RootSyncCondition, expected *v1be
 	return nil
 }
 
+// RepoSyncHasCondition returns a Predicate that errors if the RepoSync does not
+// have the specified RepoSyncCondition. Fields such as timestamps are ignored.
+func RepoSyncHasCondition(expected *v1beta1.RepoSyncCondition) Predicate {
+	return func(o client.Object) error {
+		if o == nil {
+			return ErrObjectNotFound
+		}
+		rs, ok := o.(*v1beta1.RepoSync)
+		if !ok {
+			return WrongTypeErr(rs, &v1beta1.RepoSync{})
+		}
+		condition := reposync.GetCondition(rs.Status.Conditions, expected.Type)
+		if condition == nil {
+			return fmt.Errorf("RepoSyncCondition with type %s not found", expected.Type)
+		}
+		return validateRepoSyncCondition(condition, expected)
+	}
+}
+
+func validateRepoSyncCondition(actual *v1beta1.RepoSyncCondition, expected *v1beta1.RepoSyncCondition) error {
+	e := expected.DeepCopy()
+	e.LastUpdateTime = actual.LastUpdateTime
+	e.LastTransitionTime = actual.LastTransitionTime
+	if diff := cmp.Diff(e, actual); diff != "" {
```

Comment:
`cmp.Equal(a, b, compare.IgnoreTimestampUpdates)` would work without needing to DeepCopy or change times. then diff on err.

Reply:
Done, although you do still need to change times due to the implementation of `compare.IgnoreTimestampUpdates`. I changed the function to pass by value instead of pointer

Reply:
Passing by value is problematic, because it's only a shallow copy and the k8s meta timestamps are themselves structs that contain a Time struct as a field. So without the DeepCopy here you've pointed the actual Times at the expected Times, meaning they will change in sync, which might be a gotcha if used elsewhere.

Not sure what you mean by the impl of IgnoreTimestampUpdates not working tho. Does it need to be run with cmpopts.EquateEmpty()? or does it have a bug?

Reply:
Both values must either be empty or not-empty. So since the actual value is non-empty and the expected value is empty it returns not-equal.

We could update the `expected` struct to contain an arbitrary non-nil value, but that seems less clear IMO. This is a pretty small struct so a DeepCopy doesn't seem that problematic IMO